include/pybind11/numpy.h

```diff
@@ -586,6 +586,16 @@ class dtype : public object {
         return detail::npy_format_descriptor<typename std::remove_cv<T>::type>::dtype();
     }
 
+    /// Return dtype for the given typenum (one of the NPY_TYPES).
+    /// https://numpy.org/devdocs/reference/c-api/array.html#c.PyArray_DescrFromType
+    static dtype from_typenum(int typenum) {
+        auto *ptr = detail::npy_api::get().PyArray_DescrFromType_(typenum);
```

Comment:
I thought dtype already have an explicit int ctor for this. Probably should have made it a static method in retrospect, but it's already there.

Reply:
Thanks for pointing that out, changed. The `explicit` ctor is fine IMO, I just didn't look around at all.